evalml/model_understanding/prediction_explanations/explainers.py

```diff
@@ -20,11 +20,11 @@
 )
 
 # Container for all of the pipeline-related data we need to create reports. Helps standardize APIs of report makers.
-_ReportData = namedtuple("ReportData", ["pipeline", "input_features",
+_ReportData = namedtuple("ReportData", ["pipeline", "input_features", "training_data",
                                         "y_true", "y_pred", "y_pred_values", "errors", "index_list", "metric"])
 
 
-def explain_prediction(pipeline, input_features, top_k=3, training_data=None, include_shap_values=False,
+def explain_prediction(pipeline, input_features, y=None, top_k_features=3, training_data=None, include_shap_values=False,
```

Comment:
I was thinking that we would add an argument called `index_to_explain` here and `indices_to_explain` to `explain_predictions` and have the user pass in the entire training data for `input_features`.

`_make_single_prediction_shap_table` would stay the same but we would subset the pipelines indices to these rows, e.g `pipeline_features = pipeline.compute_estimator_features(input_features).to_dataframe().iloc[indices_to_explain]`

If any of these rows contain NaNs in the features, then we raise a value error. 

I think this would be better than manually checking if the DelayedFeatureTransformer is in the pipeline and modifying the index. (We will likely add new time series transformers that modify the data in the future)